### PR TITLE
feat: add specialised replicated maps

### LINKED ISSUE #49: Add ReplicatedMultiMap
Add ReplicatedMultiMap to cover ReplicatedMap of ReplicatedSet values.

Note that in Akka (proxy impl) there is a specialized MultiMap, which isn't only ORMap of ORSet.

### PR DESCRIPTION
**Pull request by pvlugter**
Refs #48. Test the specialised replicated maps using the new TCK tests from https://github.com/lightbend/akkaserverless-framework/pull/802 (needs to be published first, draft PR here). Support the new replicated maps from https://github.com/lightbend/akkaserverless-framework/pull/803. Resolves #49.

### CODE REVIEW

File: tck/src/main/java/com/akkaserverless/javasdk/tck/model/replicatedentity/TckModelReplicatedEntity.java [hunk 1]
```diff
@@ -199,6 +201,51 @@ private void applyUpdate(ReplicatedData replicatedData, Update update) {
...
+            break;
+          case CLEAR:
+            counterMap.clear();
+            break;
+        }
+        break;
+      case REPLICATED_REGISTER_MAP:
+        @SuppressWarnings("unchecked")
+        ReplicatedRegisterMap<String, String> registerMap =
+            (ReplicatedRegisterMap<String, String>) replicatedData;
```
Comment (octonato) [inline tck/src/main/java/com/akkaserverless/javasdk/tck/model/replicatedentity/TckModelReplicatedEntity.java:229]: I may be missing something, but why both `ReplicatedCounterMap` and `ReplicatedRegisterMap` have type parameters? 

The proto definitions define the `(K, V)` to be `(String, Long)` and `(String, String)` respectively.

True that I still need to wrap my mind around the whole replicate entity feature and get a better view on how that will be used, but at first sight my understanding is that we are already bound to `(String, Long)` and `(String, String)`

Comment (pvlugter) [reply tck/src/main/java/com/akkaserverless/javasdk/tck/model/replicatedentity/TckModelReplicatedEntity.java:229]: The key can be anything serializable — protobuf message, json, or primitive. Same for register values. The proto definitions here are just for the TCK tests, which are simplified to only string keys and string values for registers.

Comment (octonato) [reply tck/src/main/java/com/akkaserverless/javasdk/tck/model/replicatedentity/TckModelReplicatedEntity.java:229]: I understood this was the TCK, but the map entry in the proto is defined as 

```proto
message ReplicatedRegisterMapEntryUpdate {
  string key = 1;
  string value = 2;
}
```

How could we send anything other then String? Are we sending the bytes encoded as base64?

Comment (pvlugter) [reply tck/src/main/java/com/akkaserverless/javasdk/tck/model/replicatedentity/TckModelReplicatedEntity.java:229]: That's the protocol for the TCK — which sends messages on how the entities should be updated, and only uses strings.

https://github.com/lightbend/akkaserverless-framework/blob/master/protocols/tck/src/main/protobuf/akkaserverless/tck/model/replicatedentity/replicated_entity.proto

The actual proxy protocol for replicated maps uses protobuf Any:

```proto
message ReplicatedMapEntryDelta {
  // The entry key.
  google.protobuf.Any key = 1;
  ReplicatedEntityDelta delta = 2;
}

message ReplicatedRegisterDelta {
  google.protobuf.Any value = 1;
  ReplicatedEntityClock clock = 2;
  int64 custom_clock_value = 3;
}
```

https://github.com/lightbend/akkaserverless-framework/blob/613d873529f8e0785218b0357c281a30a99c9d31/protocols/proxy/src/main/protobuf/akkaserverless/component/replicatedentity/replicated_entity.proto#L132-L136

https://github.com/lightbend/akkaserverless-framework/blob/613d873529f8e0785218b0357c281a30a99c9d31/protocols/proxy/src/main/protobuf/akkaserverless/component/replicatedentity/replicated_entity.proto#L119-L123

Comment (octonato) [reply tck/src/main/java/com/akkaserverless/javasdk/tck/model/replicatedentity/TckModelReplicatedEntity.java:229]: Oh, that explains everything. Thanks for clarifying.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by pvlugter** [comment]
Rebased and updated to framework release 0.7.0-beta.16.

**Pull request merged**